tcp/state.mli

```diff
@@ -40,30 +41,33 @@ type tcpstate =
   | Close_wait
   | Last_ack of Sequence.t
   | Fin_wait_1 of Sequence.t
-  | Fin_wait_2 of int
+  | Fin_wait_2
   | Closing of Sequence.t
   | Time_wait
-  | Reset
 
 val pp_tcpstate : Format.formatter -> tcpstate -> unit
 
 type close_cb = unit -> unit
 
-(* FIXME: abstract type *)
-type t = {
-  on_close: close_cb;
-  mutable state: tcpstate;
-}
+type t
 
 val state : t -> tcpstate
-val t : on_close:close_cb -> t
+val start : on_close:close_cb -> t
 
 val pp: Format.formatter -> t -> unit
 
 module Make(Time : V1_LWT.TIME) : sig
   val fin_wait_2_time : float
+  (* when in state fin_wait_2, use this as a timeout parameter  *)
   val time_wait_time : float
```

Comment:
Slightly ambiguous which value the comment refers to (`fin_wait_2_time` or `time_wait_time`).